core/tests/puppeteer-acceptance-tests/blog-admin-tests/blog-editor-create-and-delete-draft-blog-post.js

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Accpetance Test for a Blog Post Editor to create and
+ * delete draft blog posts.
+ */
+
+const e2eBlogPostEditor = require(
+  '../puppeteer-testing-utilities/blogPostAdminUtils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/testConstants.js');
+
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+const ROLE_BLOG_ADMIN = 'blog admin';
+
+let blogEditorCreateDraftAndDeleteDraftBlogPost = async function() {
+  const blogPostEditor = await new e2eBlogPostEditor();
+  await blogPostEditor.openBrowser();
+
+  await blogPostEditor.signUpNewUserWithUsernameAndEmail(
+    'blogEditor', 'testadmin@example.com');
+  await blogPostEditor.assignRoleToUser('blogEditor', ROLE_BLOG_ADMIN);
+  await blogPostEditor.expectUserToHaveRole('blogEditor', 'Blog Admin');
+
+  await blogPostEditor.goto(blogDashboardUrl);
+  await blogPostEditor.expectNumberOfDraftOrPublishedBlogPostsToBe(0);
+  await blogPostEditor.createDraftBlogPostWithTitle('Test-Blog');
+
+  await blogPostEditor.goto(blogDashboardUrl);
+  await blogPostEditor.expectDraftBlogPostWithTitleToBePresent('Test-Blog');
+
+  await blogPostEditor.deleteDraftBlogPostWithTitle('Test-Blog');
+  await blogPostEditor.expectDraftBlogPostWithTitleToBeAbsent('Test-Blog');
+
```

Comment:
Should we add `await blogPostEditor.expectNumberOfDraftOrPublishedBlogPostsToBe(0);`